### ISSUE TITLE
Linux build failed (pdf: ^3.10.3 printing: ^5.10.4)

### ISSUE THREAD

**Issue by eK14u** [DavBfr/dart_pdf#1380]
==================================================
dependencies:

  pdf: ^3.10.3
  printing: ^5.10.4

==================================================
Flutter doctor:

[✓] Flutter (Channel stable, 3.10.0, on Ubuntu 23.04 6.2.0-20-generic, locale en_US.UTF-8)
    • Flutter version 3.10.0 on channel stable at /home/erdenebayar/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 84a1e904f4 (3 days ago), 2023-05-09 07:41:44 -0700
    • Engine revision d44b5a94c9
    • Dart version 3.0.0
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /home/erdenebayar/Android/Sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: /home/erdenebayar/Dev/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 2021.2)
    • Android Studio at /home/erdenebayar/Dev/android-studio
    • Flutter plugin version 71.0.2
    • Dart plugin version 212.5744
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.1)
    • IntelliJ at /home/erdenebayar/Dev/idea-IU-221.5787.30
    • Flutter plugin version 73.1.1
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 23.04 6.2.0-20-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 113.0.5672.92

[✓] Network resources
    • All expected network resources are available.

• No issues found!
Process finished with exit code 0

==================================================
CMakeError.log:

Compiling the CXX compiler identification source file "CMakeCXXCompilerId.cpp" failed.
Compiler: /snap/flutter/current/usr/bin/clang++ 
Build flags: 
Id flags:  

The output was:
1
/snap/flutter/current/usr/bin/ld: /lib/x86_64-linux-gnu/libm.so.6: unknown type [0x13] section `.relr.dyn'
/snap/flutter/current/usr/bin/ld: skipping incompatible /lib/x86_64-linux-gnu/libm.so.6 when searching for /lib/x86_64-linux-gnu/libm.so.6
/snap/flutter/current/usr/bin/ld: cannot find /lib/x86_64-linux-gnu/libm.so.6
/snap/flutter/current/usr/bin/ld: /lib/x86_64-linux-gnu/libmvec.so.1: unknown type [0x13] section `.relr.dyn'
/snap/flutter/current/usr/bin/ld: skipping incompatible /lib/x86_64-linux-gnu/libmvec.so.1 when searching for /lib/x86_64-linux-gnu/libmvec.so.1
/snap/flutter/current/usr/bin/ld: cannot find /lib/x86_64-linux-gnu/libmvec.so.1
clang: error: linker command failed with exit code 1 (use -v to see invocation)

==================================================

Cmake version: 3.25.1

==================================================

Log:

[  +17 ms] Building Linux application...
[  +10 ms] <- compile package:manager/main.dart
[   +1 ms] executing: [build/linux/x64/debug/] cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -DFLUTTER_TARGET_PLATFORM=linux-x64 /home/erdenebayar/FlutterProjects/app-master/manager/linux
[  +30 ms] -- Downloading/updating pdfium
[  +27 ms] -- Configuring done
[   +1 ms] -- Generating done
[        ] -- Build files have been written to: /home/erdenebayar/FlutterProjects/app-master/manager/build/linux/x64/debug/pdfium-download
[  +10 ms] [1/2] Performing test step for 'pdfium-download'
[        ] FAILED: pdfium-download-prefix/src/pdfium-download-stamp/pdfium-download-test
[        ] cd /home/erdenebayar/FlutterProjects/app-master/manager/build/linux/x64/debug/pdfium-build && DOWNLOAD_EXTRACT_TIMESTAMP true && /snap/flutter/130/usr/bin/cmake -E touch
/home/erdenebayar/FlutterProjects/app-master/manager/build/linux/x64/debug/pdfium-download/pdfium-download-prefix/src/pdfium-download-stamp/pdfium-download-test
[        ] /bin/sh: 1: DOWNLOAD_EXTRACT_TIMESTAMP: not found
[        ] ninja: build stopped: subcommand failed.
[        ] CMake Error at flutter/ephemeral/.plugin_symlinks/printing/windows/DownloadProject.cmake:179 (message):
[   +1 ms]   Build step for pdfium failed: 1
[        ] Call Stack (most recent call first):
[        ]   flutter/ephemeral/.plugin_symlinks/printing/linux/CMakeLists.txt:37 (download_project)
[        ] -- Configuring incomplete, errors occurred!
[        ] See also "/home/erdenebayar/FlutterProjects/app-master/manager/build/linux/x64/debug/CMakeFiles/CMakeOutput.log".
[        ] See also "/home/erdenebayar/FlutterProjects/app-master/manager/build/linux/x64/debug/CMakeFiles/CMakeError.log".
[  +11 ms] Building Linux application... (completed in 89ms)
[+12821 ms] Exception: Unable to generate build files
[   +1 ms] "flutter run" took 13,470ms.

**Comment by DavBfr** [comment]
Fixed in the master branch.

Try it adding this in your `pubspec.yaml`
```yaml
dependency_overrides:
  printing:
    git:
      url: https://github.com/DavBfr/dart_pdf
      ref: master
      path: printing
```

**Comment by DavBfr** [comment]
Duplicates #1375

**Comment by paschalisp** [comment]
Thank you for trying to resolve this. I made a `flutter clean` and `flutter pub get` and tried build again. It downloaded `pdfium` this time, but there was an error later during the build process.

```
[   +4 ms] Launching lib/main.dart on Linux in debug mode...
[   +2 ms] /opt/software/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev /opt/software/flutter/bin/cache/dart-sdk/bin/snapshots/frontend_server.dart.snapshot --sdk-root
/opt/software/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --incremental --target=flutter --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true
-DFLUTTER_WEB_CANVASKIT_URL=https://www.gstatic.com/flutter-canvaskit/d44b5a94c976fbb65815374f61ab5392a220b084/ --output-dill /tmp/flutter_tools.RDXUDJ/flutter_tool.BZXFCA/app.dill --packages
/opt/projects/proj1/dashboard/.dart_tool/package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-scheme org-dartlang-root
--initialize-from-dill build/d2250437cd2797886fcf0d3395b1b8f5.cache.dill.track.dill --source file:///opt/projects/proj1/dashboard/.dart_tool/flutter_build/dart_plugin_registrant.dart --source
package:flutter/src/dart_plugin_registrant.dart -Dflutter.dart_plugin_registrant=file:///opt/projects/proj1/dashboard/.dart_tool/flutter_build/dart_plugin_registrant.dart --verbosity=error
--enable-experiment=alternative-invalidation-strategy
[  +19 ms] Building Linux application...
[   +7 ms] <- compile package:proj1_dashboard/main.dart
[   +4 ms] executing: [build/linux/x64/debug/] cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -DFLUTTER_TARGET_PLATFORM=linux-x64 /opt/projects/proj1/dashboard/linux
[ +233 ms] -- The CXX compiler identification is Clang 14.0.0
[   +6 ms] -- Detecting CXX compiler ABI info
[  +95 ms] -- Detecting CXX compiler ABI info - done
[   +4 ms] -- Check for working CXX compiler: /usr/bin/clang++ - skipped
[   +1 ms] -- Detecting CXX compile features
[        ] -- Detecting CXX compile features - done
[   +4 ms] -- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
[        ] -- Checking for module 'gtk+-3.0'
[  +19 ms] --   Found gtk+-3.0, version 3.24.33
[  +45 ms] -- Checking for module 'glib-2.0'
[   +6 ms] --   Found glib-2.0, version 2.72.4
[  +17 ms] -- Checking for module 'gio-2.0'
[   +7 ms] --   Found gio-2.0, version 2.72.4
[  +25 ms] -- Downloading/updating pdfium
[  +21 ms] -- Configuring done
[   +1 ms] -- Generating done
[   +3 ms] -- Build files have been written to: /opt/projects/proj1/dashboard/build/linux/x64/debug/pdfium-download
[  +52 ms] [1/9] Creating directories for 'pdfium-download'
[+1277 ms] [2/9] Performing download step (download, verify and extract) for 'pdfium-download'
[        ] -- Downloading...
[        ]    dst='/opt/projects/proj1/dashboard/build/linux/x64/debug/pdfium-download/pdfium-download-prefix/src/pdfium-linux-x64.tgz'
[        ]    timeout='none'
[        ]    inactivity timeout='none'
[        ] -- Using src='https://github.com/bblanchon/pdfium-binaries/releases/download/chromium/5200/pdfium-linux-x64.tgz'
[        ] -- [download 0% complete]
[        ] -- [download 1% complete]
[        ] -- [download 2% complete]
[        ] -- [download 3% complete]
[        ] -- [download 5% complete]
[        ] -- [download 8% complete]
[        ] -- [download 9% complete]
[        ] -- [download 11% complete]
[        ] -- [download 12% complete]
[        ] -- [download 14% complete]
[        ] -- [download 16% complete]
[        ] -- [download 18% complete]
[        ] -- [download 20% complete]
[        ] -- [download 22% complete]
[        ] -- [download 24% complete]
[        ] -- [download 25% complete]
[        ] -- [download 27% complete]
[        ] -- [download 30% complete]
[        ] -- [download 33% complete]
[        ] -- [download 35% complete]
[        ] -- [download 37% complete]
[        ] -- [download 41% complete]
[        ] -- [download 43% complete]
[        ] -- [download 45% complete]
[        ] -- [download 47% complete]
[        ] -- [download 49% complete]
[        ] -- [download 50% complete]
[        ] -- [download 52% complete]
[        ] -- [download 55% complete]
[        ] -- [download 56% complete]
[        ] -- [download 58% complete]
[        ] -- [download 59% complete]
[        ] -- [download 60% complete]
[        ] -- [download 63% complete]
[        ] -- [download 65% complete]
[        ] -- [download 67% complete]
[        ] -- [download 68% complete]
[        ] -- [download 71% complete]
[        ] -- [download 72% complete]
[        ] -- [download 73% complete]
[        ] -- [download 74% complete]
[        ] -- [download 75% complete]
[        ] -- [download 76% complete]
[        ] -- [download 78% complete]
[        ] -- [download 79% complete]
[        ] -- [download 81% complete]
[        ] -- [download 83% complete]
[        ] -- [download 85% complete]
[        ] -- [download 86% complete]
[        ] -- [download 88% complete]
[        ] -- [download 91% complete]
[        ] -- [download 93% complete]
[        ] -- [download 96% complete]
[        ] -- [download 98% complete]
[        ] -- [download 99% complete]
[        ] -- [download 100% complete]
[        ] -- Downloading... done
[        ] -- extracting...
[        ]      src='/opt/projects/proj1/dashboard/build/linux/x64/debug/pdfium-download/pdfium-download-prefix/src/pdfium-linux-x64.tgz'
[        ]      dst='/opt/projects/proj1/dashboard/build/linux/x64/debug/pdfium-src'
[        ] -- extracting... [tar xfz]
[        ] -- extracting... [analysis]
[        ] -- extracting... [rename]
[        ] -- extracting... [clean up]
[        ] -- extracting... done
[   +6 ms] [3/9] No update step for 'pdfium-download'
[  +11 ms] [4/9] No patch step for 'pdfium-download'
[  +14 ms] [5/9] No configure step for 'pdfium-download'
[  +11 ms] [6/9] No build step for 'pdfium-download'
[  +12 ms] [7/9] No install step for 'pdfium-download'
[        ] [8/9] Performing test step for 'pdfium-download'
[        ] FAILED: pdfium-download-prefix/src/pdfium-download-stamp/pdfium-download-test
/opt/projects/proj1/dashboard/build/linux/x64/debug/pdfium-download/pdfium-download-prefix/src/pdfium-download-stamp/pdfium-download-test
[        ] cd /opt/projects/proj1/dashboard/build/linux/x64/debug/pdfium-build && DOWNLOAD_EXTRACT_TIMESTAMP true && /usr/bin/cmake -E touch
/opt/projects/proj1/dashboard/build/linux/x64/debug/pdfium-download/pdfium-download-prefix/src/pdfium-download-stamp/pdfium-download-test
[        ] /bin/sh: 1: DOWNLOAD_EXTRACT_TIMESTAMP: not found
[        ] ninja: build stopped: subcommand failed.
[        ] CMake Error at flutter/ephemeral/.plugin_symlinks/printing/windows/DownloadProject.cmake:179 (message):
[   +1 ms]   Build step for pdfium failed: 1
[        ] Call Stack (most recent call first):
[        ]   flutter/ephemeral/.plugin_symlinks/printing/linux/CMakeLists.txt:36 (download_project)
[        ] -- Configuring incomplete, errors occurred!
[        ] See also "/opt/projects/proj1/dashboard/build/linux/x64/debug/CMakeFiles/CMakeOutput.log".
[  +12 ms] Building Linux application... (completed in 1,909ms)
[+19744 ms] Exception: Unable to generate build files
[   +2 ms] "flutter run" took 22,312ms.
[   +2 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:757:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1336:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:296:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:244:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:91:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:91:3)
           <asynchronous suspension>
```

Maybe there's another point that needs fixing too?

**Comment by paschalisp** [comment]
Btw, I'm using Dart 3.0.0 and Flutter 3.10.0

**Comment by DavBfr** [comment]
run `flutter pub upgrade` there is no `DOWNLOAD_EXTRACT_TIMESTAMP` anywhere in the code now.

**Comment by paschalisp** [comment]
Yes, you've right, I guessed that the package got updated since it didn't complain in the exact same point.

I did a full `flutter pub cache clean` and `flutter pub get` and `flutter pub upgrade`, however, I now run on a different build error:

```
[  +49 ms] Building Linux application...
[   +6 ms] executing: [build/linux/x64/release/] cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DFLUTTER_TARGET_PLATFORM=linux-x64 /opt/projects/proj1/dashboard/linux
[  +27 ms] -- Downloading/updating pdfium
[  +10 ms] CMake Error at CMakeLists.txt:9 (cmake_policy):
[   +1 ms]   Policy "CMP0135" is not known to this version of CMake.
[   +5 ms] -- Configuring incomplete, errors occurred!
[        ] See also "/opt/projects/proj1/dashboard/build/linux/x64/release/pdfium-download/CMakeFiles/CMakeOutput.log".
[   +1 ms] CMake Error at flutter/ephemeral/.plugin_symlinks/printing/windows/DownloadProject.cmake:171 (message):
[        ]   CMake step for pdfium failed: 1
[        ] Call Stack (most recent call first):
[        ]   flutter/ephemeral/.plugin_symlinks/printing/linux/CMakeLists.txt:36 (download_project)
[        ] -- Configuring incomplete, errors occurred!
[        ] See also "/opt/projects/proj1/dashboard/build/linux/x64/release/CMakeFiles/CMakeOutput.log".
[  +13 ms] Building Linux application... (completed in 59ms)
[        ] "flutter linux" took 611ms.
```

which perhaps is not related to `printing`, as indicates CMake policy `CMP0135` which is not known to my version of CMake (3.22.1).

I'll investigate this further and in case I find something, I'll post it here, in case it might help someone else facing the same issue.

Thanks a lot for your library!

**Comment by paschalisp** [comment]
Oh, got it already... `CMP0135` as per the [docs], is available from CMake v3.24+.

[docs]:
https://cmake.org/cmake/help/latest/policy/CMP0135.html
